### PR TITLE
feat: DPNS e2e tests

### PR DESCRIPTION
**Pull request by shumkov**
## Issue being fixed or feature implemented

In order to test DPNS contract with test suite, we need to move and slightly update DPNS functional tests from DPNS client repo https://github.com/dashevo/js-dpns-client/blob/v0.2-dev/test/functional/dpnsClient.spec.js

## What was done?

- [x] Defined test cases for DPNS
- [ ] Reworked DPNS Client functional tests into e2e and implemented all cases

## How Has This Been Tested?

Run tests

## Breaking Changes

None

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated relevant unit/integration/functional/e2e tests
- [ ] I have made corresponding changes to the documentation

**For repository code-owners and collaborators only**
- [ ] I have assigned this pull request to a milestone

### CODE REVIEW

**Comment by shumkov** [comment]
@shuplenkov @jawid-h maybe we should add more negative tests in order to check the data trigger?

**Comment by jawid-h** [comment]
@shum

> @shuplenkov @jawid-h maybe we should add more negative tests in order to check the data trigger?

Maybe it makes sense to add cases for update and delete triggers, since we're not checking them in functional tests.

**Comment by shumkov** [comment]
@jawid-h Good point! Done.

File: .env.example [hunk 1]
```diff
@@ -2,3 +2,6 @@
 DAPI_SEED=
 FAUCET_PRIVATE_KEY=
 NETWORK=
+DPNS_CONTRACT_ID=
+DPNS_TOP_LEVEL_IDENTITY=
```
Comment (shumkov) [inline .env.example:6]: DPNS_TOP_LEVEL_IDENTITY_ID=
DPNS_TOP_LEVEL_IDENTITY_PRIVATE_KEY=

File: test/e2e/dpns.spec.js [hunk 2]
```diff
@@ -1,5 +1,301 @@
...
+    expect(createdDataContract.getId()).to.equal(process.env.DPNS_CONTRACT_ID);
+  });
+
+  describe('DPNS owner', () => {
+    let createdTLD;
+    let newTopLevelDomain;
+    let replaceTopLevelDomain;
+
+    before(async () => {
+      client = await createClientWithFundedWallet(process.env.IDENTITY_MNEMONIC);
```
Comment (shumkov) [inline test/e2e/dpns.spec.js:53]: Let's rename it to `ownerClient`

File: test/e2e/dpns.spec.js [hunk 3]
```diff
@@ -1,5 +1,301 @@
...
+        const [error] = JSON.parse(e.metadata.get('errors'));
+        expect(error.name).to.equal('DataTriggerConditionError');
+        expect(error.message).to.equal('Delete action is not allowed');
+      }
+    });
+  });
+
+  describe('Any Identity', () => {
+    before(async () => {
+      client = await createClientWithFundedWallet();
```
Comment (shumkov) [inline test/e2e/dpns.spec.js:154]: we already have a client created at the top.

File: test/e2e/dpns.spec.js [hunk 4]
```diff
@@ -1,5 +1,301 @@
...
+  before(async () => {
+    topLevelDomain = 'dash';
+    secondLevelDomain = getRandomDomain();
+    client = await createClientWithFundedWallet();
+  });
+
+  after(async () => {
+    await client.disconnect();
+  });
+
```
Comment (shumkov) [inline test/e2e/dpns.spec.js:40]: describe('Data Contract')

File: .travis.yml [hunk 5]
```diff
@@ -44,14 +44,18 @@ install:
   - cd -
 
 before_script:
-  - export PRIVATE_KEY=$(mn wallet:generate-to-address local 10000 | grep -m 1 "Private key:" | awk '{printf $3}')
-  - export OPERATOR_BLS_KEY=$(mn register local $PRIVATE_KEY 127.0.0.1 20001 | grep -m 1 "Private key:" | awk '{printf $3}')
+  - export PRIVATE_KEY=$(mn wallet:generate-to-address local 10000 | grep -m 1 "Private key:" | awk '{printf $4}')
```
Comment (shumkov) [inline .travis.yml:47]: Let's use `setup-for-local-development`?

File: test/e2e/dpns.spec.js [hunk 6]
```diff
@@ -1,5 +1,305 @@
...
+  describe('Data contract', () => {
+    it('should exists', async () => {
+      const createdDataContract = await client.platform.contracts.get(process.env.DPNS_CONTRACT_ID);
+
```
Comment (shumkov) [inline test/e2e/dpns.spec.js:44]: expect(createdDataContract).to.exist()

File: test/e2e/dpns.spec.js [hunk 7]
```diff
@@ -1,5 +1,305 @@
...
+    before(async () => {
+      ownerClient = await createClientWithFundedWallet(
+        process.env.DPNS_TOP_LEVEL_IDENTITY_PRIVATE_KEY,
+      );
+
+      newTopLevelDomain = getRandomDomain();
+      replaceTopLevelDomain = getRandomDomain();
+      identity = await ownerClient.platform.identities.get(process.env.DPNS_TOP_LEVEL_IDENTITY_ID);
+
```
Comment (shumkov) [inline test/e2e/dpns.spec.js:63]: Let's add an assertion here that identity is present

File: test/e2e/dpns.spec.js [hunk 8]
```diff
@@ -1,5 +1,305 @@
...
+      const slatedDomainHashBuffer = Buffer.concat([
+        bs58.decode(preorderSalt),
+        Buffer.from(nameHash, 'hex'),
+      ]);
+
+      const saltedDomainHash = hash(
+        slatedDomainHashBuffer,
+      ).toString('hex');
+
+      const preorderDocument = await ownerClient.platform.documents.create(
```
Comment (shumkov) [inline test/e2e/dpns.spec.js:101]: Why do we need preorder here? Why not just update any `createdTLD` field?

```js
createdTLD.set('label', 123)
broadcast(replace: createdTLD)
```

File: test/e2e/dpns.spec.js [hunk 9]
```diff
@@ -1,5 +1,305 @@
...
+    });
+
+    after(async () => {
+      await ownerClient.disconnect();
+    });
+
+    // generate a random one which will be used in tests above
+    // skip if DPNS owner private key is not passed and use `dash` in tests above
+    it('should be able to register a TLD', async () => {
+      createdTLD = await ownerClient.platform.names.register(newTopLevelDomain, identity);
```
Comment (shumkov) [inline test/e2e/dpns.spec.js:74]: Let's add some assertions here? https://github.com/dashevo/platform-test-suite/pull/37/files#diff-d871dccacf05817437d39b1d0f4a692dR181-R183

File: test/e2e/dpns.spec.js [hunk 10]
```diff
@@ -1,5 +1,305 @@
...
+      const slatedDomainHashBuffer = Buffer.concat([
+        bs58.decode(preorderSalt),
+        Buffer.from(nameHash, 'hex'),
+      ]);
+
+      const saltedDomainHash = hash(
+        slatedDomainHashBuffer,
+      ).toString('hex');
+
+      const preorderDocument = await client.platform.documents.create(
```
Comment (shumkov) [inline test/e2e/dpns.spec.js:250]: We can check it without preorder

**Pull request merged**